libraries/ruby_install.rb

```diff
@@ -0,0 +1,247 @@
+#
+# Cookbook Name:: omnibus
+# HWRP:: ruby_install
+#
+# Copyright 2014, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require_relative '_helper'
+
+class Chef
+  class Resource::RubyInstall < Resource::LWRPBase
+    self.resource_name = :ruby_install
+
+    actions :install
+    default_action :install
+
+    attribute :version,     kind_of: String, name_attribute: true
+    attribute :environment, kind_of: Hash, default: {}
+    attribute :patches,     kind_of: Array, default: []
+
+    def patch(patch)
+      @patches << patch
+    end
+  end
+
+  class Provider::RubyInstallUnix < Provider::LWRPBase
+    provides :ruby_install
+
+    def whyrun_supported?
+      true
+    end
+
+    action(:install) do
+      if installed?
+        Chef::Log.debug("#{new_resource} installed - skipping")
+      else
+        converge_by("install #{new_resource}") do
+          install
+        end
+      end
+    end
+
+    protected
+
+    def version
+      new_resource.version
+    end
+
+    def compile_flags
+      [
+        '--disable-install-rdoc',
+        '--disable-install-ri',
+        '--with-out-ext=tcl',
+        '--with-out-ext=tk',
+        '--without-tcl',
+        '--without-tk',
+        '--disable-dtrace',
+      ].join(' ')
+    end
+
+    def install
+      install_dependencies
+
+      # Need to compile the command outside of the execute resource because
+      # Ruby is bad at instance_eval
+      install_command = 'ruby-install --no-install-deps'
```

Comment:
Assuming you add the `prefix` attribute [mentioned above](https://github.com/chef-cookbooks/languages/pull/4/files#r42443869) this command would change to be something like:

```ruby
install_command = "ruby-install --no-install-deps --install-dir #{new_resource.prefix}" 
```